### PR TITLE
Document JsonApiEndpoints enum

### LINKED ISSUE #1298: JsonApiEndpoints documentation
Hi, it's a bit unclear what each value of the JsonApiEndpoints does. For example, I might be missing some JSON:API concept but I'm not sure what `GetSecondary` is. Would it possible to add XML comments on those? 🙏Ideally, an example endpoint for each would be great.

### PR DESCRIPTION
**Pull request by verdie-g**
Closes #1298

#### QUALITY CHECKLIST
- [X] Changes implemented in code
- [X] Complies with our [contributing guidelines](https://github.com/json-api-dotnet/JsonApiDotNetCore/blob/master/.github/CONTRIBUTING.md)
- [X] Adapted tests
- [X] Documentation updated

### CODE REVIEW

**Comment by bkoelman** [comment]
Thanks. I'll get back to this next month, after my break.

**Review by bkoelman — CHANGES_REQUESTED** [review]
Please note the line at the top of this file:
```
// IMPORTANT: An internal copy of this type exists in the SourceGenerators project. Keep these in sync when making changes.
```

These changes should be replicated in the other file.

File: src/JsonApiDotNetCore.Annotations/Controllers/JsonApiEndpoints.shared.cs [hunk 1]
```diff
@@ -10,20 +10,104 @@ namespace JsonApiDotNetCore.Controllers;
...
+    /// GET /articles/1 HTTP/1.1
+    /// ]]></code>
+    /// </summary>
     GetSingle = 1 << 1,
+
+    /// <summary>
+    /// Represents the endpoint to get a secondary resource or collection of secondary resources. Example:
+    /// <code><![CDATA[
+    /// GET /articles/1/author HTTP/1.1
+    /// ]]></code>
```
Comment (bkoelman) [inline src/JsonApiDotNetCore.Annotations/Controllers/JsonApiEndpoints.shared.cs:36]: Did you intentionally leave out the second example? See https://github.com/json-api-dotnet/JsonApiDotNetCore/blob/master/src/JsonApiDotNetCore/Controllers/BaseJsonApiController.cs#L126-L133.

File: src/JsonApiDotNetCore.Annotations/Controllers/JsonApiEndpoints.shared.cs [hunk 2]
```diff
@@ -10,20 +10,104 @@ namespace JsonApiDotNetCore.Controllers;
...
+    /// GET /articles/1/relationships/author HTTP/1.1
+    /// ]]></code> Example:
+    /// <code><![CDATA[
+    /// GET /articles/1/relationships/revisions HTTP/1.1
+    /// ]]></code>
+    /// </summary>
     GetRelationship = 1 << 3,
+
+    /// <summary>
+    /// Represents the endpoint to creates a new resource with attributes, relationships or both. Example:
```
Comment (bkoelman) [inline src/JsonApiDotNetCore.Annotations/Controllers/JsonApiEndpoints.shared.cs:51]:
```suggestion
    /// Represents the endpoint to create a new resource with attributes, relationships or both. Example:
```

**Review by bkoelman — CHANGES_REQUESTED** [review]

**Comment by verdie-g** [comment]
The CI error is weird, I did run cleanupcode 🤔

**Comment by bkoelman** [comment]
> The CI error is weird, I did run cleanupcode 🤔

Strange indeed. Running locally on Windows does not produce a diff. I haven't seen this happen before, will investigate. What OS are you using?

**Comment by verdie-g** [comment]
Windows 11 and
```
$ dotnet --info
.NET SDK:
 Version:   7.0.400
 Commit:    73bf45718d

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.400\

Host:
  Version:      7.0.10
  Architecture: x64
  Commit:       a6dbb800a4

.NET SDKs installed:
  6.0.413 [C:\Program Files\dotnet\sdk]
  7.0.400 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.21 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found
```

**Comment by verdie-g** [comment]
Could it be a line-ending thing? Maybe locally I'm using CRLF and the CI is using LF the number of characters on the line is different?

**Comment by bkoelman** [comment]
> Could it be a line-ending thing? Maybe locally I'm using CRLF and the CI is using LF the number of characters on the line is different?

Yes, but I don't think that should matter. It has always worked that way. I've created a temporary PR #1304 to investigate.

**Comment by bkoelman** [comment]
@verdie-g Sorry this is taking so long. I've opened a bug at https://youtrack.jetbrains.com/issue/RSRP-494002, but it hasn't been triaged yet.

Meanwhile, can you please merge or rebase on the latest changes in master? I've replaced AppVeyor with GitHub Actions. This is a nice opportunity to test whether everything works from a community PR. I can't do it myself because my account has write permissions and I'd like to confirm builds work without that. Sadly, it won't fix the cleanupcode bug though.

**Comment by verdie-g** [comment]
Looks like it's failing for macos

**Comment by bkoelman** [comment]
> Looks like it's failing for macos

Yes, same as linux.

**Comment by bkoelman** [comment]
@verdie-g I'd like to keep this PR open. I suspect this was closed unintentionally as a side effect of deleting your fork. I'me unable to reopen because your fork no longer exists. Can you restore it?

We still want to take this PR. The related issue has been triaged with priority critical, so I have high hopes a fix will become available soon.

**Comment by verdie-g** [comment]
I completely forgot about that PR. I'll try restoring the fork when I get some time.

**Comment by bkoelman** [comment]
Superseded by #1423.

**Pull request closed without merging**